### ISSUE TITLE
Reproducing zero-shot results on ESC50

### ISSUE THREAD

**Issue by Tom-Huang** [LAION-AI/CLAP#60]
Dear authors of CLAP, 
I am trying to reproduce the zero-shot classification results on the ESC50 dataset shown in the paper. I tried the script in the repo: [https://github.com/LAION-AI/CLAP/blob/main/experiment_scripts/zeroshot_esc50.sh](https://github.com/LAION-AI/CLAP/blob/main/experiment_scripts/zeroshot_esc50.sh) and I also downloaded the ESC50 dataset in webdata format as in [https://drive.google.com/drive/folders/1aU54FGctrjhxA2sTN0wgHVsPm0nPEc_E](https://drive.google.com/drive/folders/1aU54FGctrjhxA2sTN0wgHVsPm0nPEc_E). I only do the following modifications in the zeroshot_esc50.sh script:
* I removed `--remotedata` and added `--datasetpath "/path/to/downloaded/esc50/folder"`
* I changed `--openai-model-cache-dir` to a customized path
* I changed `--pretrained` to the dir of the downloaded checkpoint for **Model train on LAION-Audio-630K + Audioset (Keyword to caption)**. The dir contains `epoch_top_0.pt` while `out.log` and `params.txt` are sibling files of the dir.

The modified version of the script can be found here: [https://drive.google.com/file/d/1wPU0sojzKQc313QtXil03NqxLxqeOzfW/view?usp=sharing](https://drive.google.com/file/d/1wPU0sojzKQc313QtXil03NqxLxqeOzfW/view?usp=sharing)

After running this script, I got very low classification accuracy. The log is here: [https://drive.google.com/file/d/1C9d8WcezRVfwtBV5utpZEAE27Q4DXBmA/view?usp=sharing](https://drive.google.com/file/d/1C9d8WcezRVfwtBV5utpZEAE27Q4DXBmA/view?usp=sharing).

I am wondering if I am using the correct model or if I need extra modifications to get the results shown in the paper. Maybe you could provide some hints to me. Thank you in advance!

**Comment by RetroCirce** [comment]
Hi Tom,

Thank you for your post, now we are making our laion_clap as a pip library, with an ESC-50 evaluation tutorial (it should be released a week later). 
We did find one ckpt in our google drive is wrong (as we upload a wrong name model). We will fix it as we releaase our laioh_clap pip library

Best,
Ke

**Comment by RetroCirce** [comment]
Now we have updated our repo, release our pip library, and a tutorial code to evaluate it in the ESC-50 dataset. You can track our readme.md to use it.

**Comment by Tom-Huang** [comment]
Thank you very much! I will try the library code!

**Comment by CoderwayNew** [comment]
I've read the paper and the instructions on github. But I did not find what openai-model-cache-dir is used for. May I ask what --openai-model-cache-dir is used to store? 
Thanks in advance!

**Comment by lukewys** [comment]
Hi @CoderwayNew , the --openai-model-cache-dir is for downloading the CLIP checkpoint if you want to use the CLIP text encoder and the text encoder. It is not used in the experiment and we found it not perform very good.
